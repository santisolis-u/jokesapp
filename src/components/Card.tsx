import React from 'react';
import {Animated, StyleSheet, Text, Dimensions} from 'react-native';

const {width} = Dimensions.get('window');

interface JokeCardProps {
  text: string;
  backgroundColor: string;
}

const JokeCard = ({text, backgroundColor}: JokeCardProps) => {
  return (
    <Animated.View style={[styles.card, {backgroundColor}]}>
      <Text style={[styles.text]}>{text}</Text>
      <Text style={[styles.author]}>@designhumor</Text>
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  card: {
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingTop: 70,
    paddingBottom: 30,
    paddingHorizontal: 20,
    borderRadius: 8,
    marginHorizontal: 20,
    width: width * 0.8,
  },
  text: {
    color: 'white',
    textAlign: 'center',
    fontSize: 24,
    marginHorizontal: 20,
    fontFamily: 'OpenSans-Bold',
  },
  author: {
    fontSize: 22,
    color: 'white',
    fontFamily: 'IndieFlower',
  },
  bottom: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  authorWithIcon: {
    marginRight: 8,
  },
});

export default JokeCard;
