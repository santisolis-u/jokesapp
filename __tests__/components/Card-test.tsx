import 'react-native';
import React from 'react';

import Card from '../../src/components/Card';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Renders', () => {
  it('Renders correctly', () => {
    const json = renderer
      .create(<Card isLoading={false} text={'test'} backgroundColor={'red'} />)
      .toJSON();
    expect(json).toMatchSnapshot();
  });
  it('Renders correctly loading', () => {
    const json = renderer
      .create(
        <Card isLoading={true} text={'testing'} backgroundColor={'red'} />,
      )
      .toJSON();
    expect(json).toMatchSnapshot();
  });
});
