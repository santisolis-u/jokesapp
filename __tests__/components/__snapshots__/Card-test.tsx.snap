// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Renders correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "red",
      "borderRadius": 8,
      "justifyContent": "space-between",
      "marginHorizontal": 37.5,
      "paddingBottom": 30,
      "paddingHorizontal": 20,
      "paddingTop": 70,
      "width": 600,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 24,
          "marginHorizontal": 20,
          "textAlign": "center",
        },
        null,
      ]
    }
  >
    test
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "IndieFlower",
          "fontSize": 22,
        },
        null,
      ]
    }
  >
    @designhumor
  </Text>
</View>
`;

exports[`Renders Renders correctly loading 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(237, NaN, NaN, 0.7)",
      "borderRadius": 8,
      "justifyContent": "space-between",
      "marginHorizontal": 37.5,
      "paddingBottom": 30,
      "paddingHorizontal": 20,
      "paddingTop": 70,
      "width": 600,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "marginTop": 50,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <ActivityIndicator
      color="black"
      size="large"
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 24,
          "marginHorizontal": 20,
          "textAlign": "center",
        },
        Object {
          "opacity": 0.7,
        },
      ]
    }
  >
    testing
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "IndieFlower",
          "fontSize": 22,
        },
        Object {
          "opacity": 0.7,
        },
      ]
    }
  >
    @designhumor
  </Text>
</View>
`;
